### ISSUE TITLE
OpenId token not refreshed if the config is in-memory

### ISSUE THREAD

**Issue by ttbadr** [fabric8io/kubernetes-client#5880]
As title, If we use a in-memory config, the token not refreshed. 
I found the method below, it only save the new token when the config is file
https://github.com/fabric8io/kubernetes-client/blob/a741e4b8fb5908c584813da53c7eaddc638dd81b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/OpenIDConnectionUtils.java#L107

**Comment by rohanKanojia** [comment]
@ttbadr : Could you please provide more details? How are OpenID parameters being provided to KubernetesClient (like refresh_token, OpenID identity provider URL, etc)?

**Comment by ttbadr** [comment]
@rohanKanojia I think the root cause is [here](https://github.com/fabric8io/kubernetes-client/blob/a741e4b8fb5908c584813da53c7eaddc638dd81b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/OpenIDConnectionUtils.java#L245)
this method `persistKubeConfigWithUpdatedAuthInfo` will be called when the refresh token request success, and persist the new token to the config.
but you can see that if the config is not a file then return, if the config is a file then save the new token to the kubeconfig. 
so if the config is in-memory, the new token will be ignore, the next client request will use the old token and fail

**Comment by rohanKanojia** [comment]
@ttbadr : May I know which cluster you're using? It might be difficult for us to reproduce this. Is it possible for you to create a pull request to fix this?

**Comment by ttbadr** [comment]
@rohanKanojia I use the k8s, I create a pull request to fix it #5888, can you help to review it. thx

**Comment by rohanKanojia** [comment]
@ttbadr : Could you please add a test case to validate your fix?

**Comment by ttbadr** [comment]
@rohanKanojia ok, I can add some test cases. Sorry I can't provide the source, It's a internal k8s cluster

**Comment by rohanKanojia** [comment]
@ttbadr : I'm not asking for your cluster details. I'm requesting you to elaborate more on your problem. How is Config loaded by KubernetesClient? Is it via some local `.kubeconfig` file or via ConfigBuilder?

**Comment by ttbadr** [comment]
@rohanKanojia I build the Kubeclient via ConfigBuilder, the code like below:
```java
Config config = new ConfigBuilder().withAutoConfigure(false)
        .withNameSpace(namespace)
        .withMasterUrl(url)
        .withAutoOAuthToken(token)
        .withCaCertData(base64CaData)
        .withAuthProvider(new AuthProviderConfigBuilder()
                .withName(providerName)
                .withConfig(configMap)
                .build()).build();
new KubernetesClientBuilder().withConfig(config).build();
```
btw, I can't find the ConfigBuilder class in the source, is it genarated by some maven task?

**Comment by rohanKanojia** [comment]
@ttbadr: Yes, it's generated using [sundrio annotation processor ](https://github.com/sundrio/sundrio)

**Comment by manusa** [comment]
After working on the refactor of the OpenIDConnectionUtils, I'm noticing that these changes might actually create a regression for #4802 which was fixed by #4951

Please, @Vlatombe, could you review #5888 and verify it won't cause a regression for your use cases.
